docker-compose.yaml

```diff
@@ -0,0 +1,78 @@
+name: photoview
+
+volumes:
+  mariadb:
+  postgres:
```

Comment:
Yeah. I agree. But `mysql` and `postgres` change uid when running, making the ownership of local files complex, especially on macOS, as the docker runs in a VM instead of the host.

If you have a better idea about how to mount a volume in the compose on macOS, let me know.

Reply:
I work on Mac as well, and in all my compose files I have direct explicit mappings from the product config folder on the host to the path inside the container (like in the compose examples).
Yes, DB containers create and access files in volumes under root, but that is not an issue for me, as I always could run `sudo chmod -R 777 ./<volume's folder>` command on the host before accessing those files

Reply:
I can't do that with `mariadb:lts`. `mariadb` switches `/var/lib/mysql` to `mysql` as the owner, which id is `999`.

```bash
% ls -lh ./dev/                                                                                          
total 0
drwxr-xr-x  2 999  staff    64B Jul  1 20:47 mysql
✓20:48:53 [googol@Googol-MBA:~/code/photoview] on scripts
% docker run --rm -v `pwd`/dev/mysql:/var/lib/mysql -e MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=true mariadb:lts id -u mysql
999
✓20:49:06 [googol@Googol-MBA:~/code/photoview] on scripts
% docker run --rm -v `pwd`/dev/mysql:/var/lib/mysql -e MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=true mariadb:lts            
2024-07-01 18:49:11+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:11.4.2+maria~ubu2404 started.
2024-07-01 18:49:11+00:00 [Warn] [Entrypoint]: /sys/fs/cgroup///memory.pressure not writable, functionality unavailable to MariaDB
2024-07-01 18:49:11+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2024-07-01 18:49:11+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:11.4.2+maria~ubu2404 started.
2024-07-01 18:49:12+00:00 [Note] [Entrypoint]: Initializing database files
2024-07-01 18:49:12 0 [Warning] Can't create test file '/var/lib/mysql/f1ad977e317d.lower-test' (Errcode: 13 "Permission denied")
2024-07-01 18:49:12 0 [ERROR] mariadbd: Can't create/write to file './ddl_recovery.log' (Errcode: 13 "Permission denied")
2024-07-01 18:49:12 0 [ERROR] DDL_LOG: Failed to create ddl log file: ./ddl_recovery.log
2024-07-01 18:49:12 0 [ERROR] Aborting
...
```

Reply:
sorry, didn't really get the problem: you have an empty `pwd`/dev/mysql folder at the beginning, owned by 999:staff. Then you run the DB container with mounting that folder as a volume and specifying the user, and there is no output.
Finally, you run the same command, but with no user param and get the access denied error.

But I don't see where you run the `chmod 777` on that folder, and why exactly that wouldn't work for you?
The `chmod` command simply grants all permissions to all users for that folder and all its content (with -R param). Of course, all newly created by MariaDB files would be still not accessible, so you'd need to re-run the `chmod` command periodically, but I don't see any negative impact, as you don't change ownership or anyhow decrease the originally granted permissions. You just add the missing permissions.

Reply:
Well, I hope to avoid `777`, even in dev env.

Added `scripts/up_database.sh` to do that.